flink-test-utils-parent/flink-migration-test-utils/src/main/java/org/apache/flink/test/migration/PublishedVersionUtils.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.migration;
+
+import org.apache.flink.FlinkVersion;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Objects;
+
+/** Utilities for update and read the most-recently published flink version. */
+public class PublishedVersionUtils {
+
+    private static final String MOST_RECENTLY_PUBLISHED_VERSION_FILE =
+            "most_recently_published_version";
+
+    public static FlinkVersion getMostRecentlyPublishedVersion() {
```

Comment:
I'm wondering whether we should add some functionality for ignoring comments (e.g. lines starting with an `#`). That way we could add some comment into the file to give some context for the value. Just as an idea: You don't have to do it if you think that it's to much...

Reply:
It should indeed make the value more meaningful, It looks to me perhaps we could make it a follow-up optimization, for now other developers could locate the meaning of the file via the git history, and users should not be able to notice this file.